### PR TITLE
File name bug when editing the config #2

### PR DESCRIPTION
**Pull request by ForlaxPy**
Fixed it so it works for all the configs in all subs dir

### CODE REVIEW

File: OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs [hunk 1]
```diff
@@ -148,9 +149,16 @@ public void SaveConfig()
             Current.Config.Settings.LastModified = DateTime.Now;
             Current.Config.Settings.Version = Globals.obVersion;
             Globals.LogInfo(Components.ConfigManager, "Converted the unbinded observables and set the Last Modified date");
-            
+
+            if (File.Exists(Current.Path) && ((Current.Config.Settings.Name + ".loli") != Current.Path.Replace(Globals.configFolder + "\\", "")))
```
Comment (openbullet) [inline OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs]: You can simply do
```csharp
Current.Config.Settings.Name != Path.GetFileNameWithoutExtension
```

File: OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs [hunk 2]
```diff
@@ -148,9 +149,16 @@ public void SaveConfig()
...
+            {
+                string newpath = Globals.configFolder + "\\" + (Current.Category == "Default" ? "" : (Current.Category + "\\")) + Current.Config.Settings.Name + ".loli";
```
Comment (openbullet) [inline OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs]: You can combine paths with the `Path.Combine()` method. Do not reinvent the wheel mate!
e.g.
```csharp
string newPath = Path.Combine(Globals.configFolder, Current.Category, Current.Config.Settings.Name + ".loli");
```

File: OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs [hunk 3]
```diff
@@ -148,9 +149,16 @@ public void SaveConfig()
...
+                File.Move(Current.Path, newpath);
```
Comment (openbullet) [inline OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs]: What if the new config name has invalid windows characters like < or : or > ?
You didn't take account for those. I made a static method in the BlockBase class called `MakeValidFileName` you NEED to call it before doing this if you don't want the program to crash upon bad characters!

**Comment by openbullet** [comment]
> aren't checkpoints also based on the config + wordlist names? So you have to rename the checkpoint keys in DB too.
It's a pain in the ass to do this, I really didn't want to work on this feature for a reason.

**Comment by ForlaxPy** [comment]
> 
> 
> Hmm... my 2 cents on this.
> Basically a good idea.
> 
> But I've seen some similar request on some other software.
> Where the filename changed with some project name.
> And this caused problems with forbidden device names, pipe and other special characters even max lenght of path (ok with modern Windows not such a problem anymore. But maybe with networkstorage)
> 
> And for OB: aren't checkpoints also based on the config + wordlist names? So you have to rename the checkpoint keys in DB too.
> And maybe other problems i can't think of right now

shiot I didn't know that these stuff were linked together I think this is going to be causing problems. Anyway I see the point why it wasn't added. @openbullet Ruri up to you, you could close it now and I will be fully convering it tomorrow or if you like to keep working on it then go ahead, I already moved to some others issues scripts so will have to finish em before coming back to this one.

File: OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs (hunk 2 above)
Comment (openbullet) [reply OpenBullet/Pages/Main/Configs/ConfigManager.xaml.cs]: Another way to do this (much easier and that automatically accounts for all the subfolder or no subfolder) is this one:
```csharp
string newPath = Path.Combine(Path.GetDirectoryName(Current.Path), Current.Config.Settings.Name);
```

**Comment by openbullet** [comment]
@ForlaxPy I think if done correctly this can work decently, even if we don't fix the checkpoints it's still fine, just fix the stuff that I pointed out in the code review and then I'll merge it!

**Comment by openbullet** [comment]
@ForlaxPy by the way I'll keep code reviewing every PR you make, don't take it personally, I just want new code to be in line with the existing codebase and I want to take account for any possible consequence of adding a new feature as to not introduce new bugs!

**Comment by meinname** [comment]
yeah - it took you more than 2 weeks or so for checking in my TCP thing :P

**Comment by ForlaxPy** [comment]
Yeah ofc I do understand you no need to clarify ^^.

**Pull request closed without merging**